galsim/bandpass.py

```diff
@@ -211,28 +229,17 @@ def __div__(self, other):
             red_limit = min([self.red_limit, other.red_limit])
             wave_list = wave_list[(wave_list >= blue_limit) & (wave_list <= red_limit)]
 
-        # product of Bandpass instance and Bandpass subclass instance
-        if isinstance(other, Bandpass) and type(self) != type(other):
-            ret = Bandpass(lambda w: self(w)/other(w),
-                           blue_limit=blue_limit, red_limit=red_limit,
-                           _wave_list=wave_list)
-        # otherwise, preserve type of self
+        if hasattr(other, '__call__'):
+            tp = lambda w: self.func(w) / other(w)
         else:
-            ret = self.copy()
-            ret.blue_limit = blue_limit
-            ret.red_limit = red_limit
-            ret.wave_list = wave_list
-            ret.zeropoint = None
-            if hasattr(ret, '_effective_wavelength'):
-                del ret._effective_wavelength # this will get lazily recomputed when needed
-            if hasattr(other, '__call__'):
-                ret.func = lambda w: self(w)/other(w)
-            else:
-                ret.func = lambda w: self(w)/other
-        return ret
+            tp = lambda w: self.func(w) / other
+
+        return Bandpass(tp, blue_limit, red_limit, _wave_list=wave_list)
 
     # Doesn't check for divide by zero, so be careful.
     def __rdiv__(self, other):
+        # MJ: I can't think what the use case is here.  Why would we want to divide 
+        #     something (scalar or funct) by a bandpass?
         blue_limit = self.blue_limit
```

Comment:
My thought here was that one might have a bandpass describing the total system throughput, and another describing the atmosphere throughput.  One could then divide them to get the "above-the-atmosphere" transmission function, which might be a useful photometric calibration tool.  Probably the opposite (multiplying to obtain the total system throughput) is more likely in practice, but I didn't think it hurt to have this just in case.